hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/transfer/AdjustFungibleTokenChangesStep.java

```diff
@@ -90,6 +90,11 @@ public void doIn(@NonNull final TransferContext transferContext) {
                 getIfUsable(accountId, accountStore, handleContext.expiryValidator(), INVALID_ACCOUNT_ID);
                 final var pair = new EntityIDPair(accountId, tokenId);
 
+                // Validate freeze status and kyc granted
+                final var accountID = aa.accountIDOrThrow();
+                final var tokenRel = getIfUsable(accountID, tokenId, tokenRelStore);
+                validateNotFrozenAndKycOnRelation(tokenRel);
```

Comment:
Great catch. 🙌

Reply:
Thanks! We found it with @mustafauzunn 🙌